modules/caddyhttp/reverseproxy/selectionpolicies.go

```diff
@@ -387,6 +375,113 @@ func (s *HeaderHashSelection) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 	return nil
 }
 
+// CookieHashSelection is a policy that selects
+// a host based on a given cookie name.
+type CookieHashSelection struct {
+	// The HTTP cookie name whose value is to be hashed and used for upstream selection.
+	Name string `json:"name,omitempty"`
+	// Secret to hash (Hmac256) chosen upstream in cookie
+	Secret string `json:"secret,omitempty"`
+}
+
+// CaddyModule returns the Caddy module information.
+func (CookieHashSelection) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "http.reverse_proxy.selection_policies.cookie",
+		New: func() caddy.Module { return new(CookieHashSelection) },
+	}
+}
+
+// Select returns an available host, if any.
+func (s CookieHashSelection) Select(pool UpstreamPool, req *http.Request, w http.ResponseWriter) *Upstream {
+	if s.Name == "" {
+		s.Name = "lb"
+	}
+	cookie, err := req.Cookie(s.Name)
+	// If there's no cookie, select new random host
+	if err != nil || cookie == nil {
+		return selectNewHostWithCookieHashSelection(pool, w, s.Secret, s.Name)
+	} else {
+		// If the cookie is present, loop over the available upstreams until we find a match
+		cookieValue := cookie.Value
+		for _, upstream := range pool {
+			if !upstream.Available() {
+				continue
+			}
+			sha, err := hashCookie(s.Secret, upstream.Dial)
+			if err == nil && sha == cookieValue {
+				return upstream
+			}
+		}
+	}
+	// If there is no matching host, select new random host
+	return selectNewHostWithCookieHashSelection(pool, w, s.Secret, s.Name)
+}
+
+// UnmarshalCaddyfile sets up the module from Caddyfile tokens. Syntax:
+// lb_policy cookie [<name> [<secret>]]
```

Comment:
Typically this is indented 4 spaces and given buffer lines because it can be rendered as markdown (code blocks) by some tooling

```suggestion
//
//     lb_policy cookie [<name> [<secret>]]
//
```

Reply:
Ok, thanks for the info !